sys/isrpipe/isrpipe.c

```diff
@@ -0,0 +1,29 @@
+#include "isrpipe.h"
+
+void isrpipe_init(isrpipe_t *isrpipe, char *buf, size_t bufsize)
+{
+    mutex_init(&isrpipe->mutex);
+    tsrb_init(&isrpipe->tsrb, buf, bufsize);
+}
+
+int isrpipe_write_one(isrpipe_t *isrpipe, char c)
+{
+    int res = tsrb_add_one(&isrpipe->tsrb, c);
+
+    /* `res` is either 0 on success or -1 when the buffer is full. Either way,
+     * unlocking the mutex is fine.
+     */
+    mutex_unlock(&isrpipe->mutex);
+
+    return res;
+}
+
+int isrpipe_read(isrpipe_t *isrpipe, char *buffer, size_t count)
+{
+    int res;
+
+    while (!(res = tsrb_get(&isrpipe->tsrb, buffer, count))) {
```

Comment:
I had implemented those methods myself in the past and I faced one issue with a [very] slow uart device. 
The code for `tsrb_get` is:

```C
int tsrb_get(tsrb_t *rb, char *dst, size_t n)
{
    size_t tmp = n;
    while (tmp && !tsrb_empty(rb)) {
        *dst++ = _pop(rb);
        tmp--;
    }
    return (n - tmp);
}
```

Imagine you want to read 5 chars but there's only 1 available at a time. 
First time you call `tsrb_get` the `rb` is empty so you get 0 and you lock. A single character arrives, the mutex is unlocked and you call `tsrb_get` again: now it returns 1, so you don't lock anymore and you return 1 single byte. Also could happen that there's a leftover byte in the uart when you call the read method: same thing.
Is this the expected behaviour? 

Maybe this could be tackled by defining two different methods: one would have this behaviour and one would block until 5 bytes were really available. Or maybe the documentation updated. Or am I missing something...?